toolchain/check/handle_struct.cpp

```diff
@@ -61,6 +119,10 @@ auto HandleStructLiteral(Context& context, Parse::Node parse_node) -> bool {
       .PopAndDiscardSoloParseNode<
           Parse::NodeKind::StructLiteralOrStructTypeLiteralStart>();
   auto type_block_id = context.args_type_info_stack().Pop();
+  if (HasDuplicateNamesError(context, type_block_id)) {
```

Comment:
To me, this name isn't clear that it diagnoses the error if it finds it. Something like `CheckForDuplicateNames` or `DiagnoseDuplicateNames` would be clearer to me.